### DISCUSSION TITLE
Cert error in AWS lambda for .net core 8 SSL_CERT_DIR

### DISCUSSION THREAD

**Discussion by sajanmje** [dotnet/runtime#101274] [Q&A] [answered]
We're attempting to retrieve a list of secrets from the AWS Parameter Store using .NET Core 8. To make external API calls, we included a certificate in the Lambda's environment variables with the **key** SSL_CERT_DIR and the **value** /opt/certs. However, after adding this variable, we encountered a certificate error while retrieving data from the parameter store. It worked fine without the SSL_CERT_DIR environment variable. How can we resolve this issue?

.NET 8 uses Amazon Linux 2023 which has OpenSSL 3 versus .NET 6 used Amazon Linux 2 and OpenSSL 1

**Comment by Boxter187** [comment]
> We're attempting to retrieve a list of secrets from the AWS Parameter Store using .NET Core 8. To make external API calls, we included a certificate in the Lambda's environment variables with the **key** SSL_CERT_DIR and the **value** /opt/certs. However, after adding this variable, we encountered a certificate error while retrieving data from the parameter store. It worked fine without the SSL_CERT_DIR environment variable. How can we resolve this issue?
> 
> .NET 8 uses Amazon Linux 2023 which has OpenSSL 3 versus .NET 6 used Amazon Linux 2 and OpenSSL 1

**Comment by tmds** [comment, marked as answer]
`SSL_CERT_DIR` can be set to a colon separated list of directories.

Try setting it to `/etc/ssl/certs:/opt/certs` to continue including the default location.

If that doesn't work, you should check if `SSL_CERT_DIR` isn't already set by AWS (when you have not provided a value) to include non-default locations. If so, you should include them as well.
